gecko/glue/include/MediaSource.h

```diff
@@ -0,0 +1,144 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_MediaSource_h_
+#define mozilla_dom_MediaSource_h_
+
+#include "MediaSourceDecoder.h"
+#include "mozilla/Assertions.h"
+#include "mozilla/Attributes.h"
+#include "mozilla/MozPromise.h"
+#include "mozilla/dom/MediaSourceBinding.h"
+#include "nsCOMPtr.h"
+#include "nscore.h"
+#include "TimeUnits.h"
+
+namespace mozilla {
+
+// TODO (gecko-media) class AbstractThread;
```

Comment:
I believe we have AbstractThread in gecko-media already. Why was it disabled in this header?

Reply:
Because Gecko [gets it from the global object](https://dxr.mozilla.org/mozilla-central/source/dom/media/mediasource/MediaSource.cpp#497) and I need to figure out what that means for us :)

Reply:
In any case, as @cpearce mentioned [above](https://github.com/servo/gecko-media/pull/81#issuecomment-345644525) I need to rewrite MediaSource, so I'm not sure how much of this code will be left.

Reply:
Upstream Gecko gets the Abstract MainThread from the script global object so that that runnables dispatched to the thread can be dispatched in the correct order when Gecko has multiple threads which correspond to the gecko "main" thread. There's [more details here](https://wiki.mozilla.org/Quantum/DOM), but since in gecko-media we're aiming to have a single thread that thinks it's the Gecko main thread, we can just use AbstractThread::MainThread(), [as we've already done in a few places](https://github.com/servo/gecko-media/search?utf8=%E2%9C%93&q=AbstractThread%3A%3AMainThread%28%29&type=) where we need an AbstractThread representation of the main thread.

Reply:
Makes sense. Thanks for the explanation!